src/reducers/AddSiteReducer.js

```diff
@@ -117,6 +131,12 @@ export default (state = INITIAL_STATE, action) => {
                 readyLongitude: longitude
             };
 
+        case SITE_DETAIL_CHECKBOX_UPDATED:
+            const {siteDetailCheckboxKey} = payload;
+            const updatedSiteDetailCheckboxesKeys = updateSiteDetailCheckboxesKeys(state, siteDetailCheckboxKey);
+
+            return {...state, siteDetailCheckboxesKeys: updatedSiteDetailCheckboxesKeys};
```

Comment:
Similar to our logic in `FilterScreen`. When a new checkbox payload comes in, fire off to `updateSiteDetailCheckboxesKeys` to get the latest list of what's checked.